js/source/geojson_source_worker.js

```diff
@@ -0,0 +1,98 @@
+'use strict';
+
+var ajax = require('../util/ajax');
+var rewind = require('geojson-rewind');
+var GeoJSONWrapper = require('./geojson_wrapper');
+var vtpbf = require('vt-pbf');
+var geojsonvt = require('geojson-vt');
+
+/*
+ * A base prototype to use for creating GeoJSON-based source workers.  This is
+ * basically an abstract class; subclasses should implement (at least) the
+ * indexData method.
+ */
```

Comment:
Do you think we could implement a robust plugin system using interfaces and composition rather than inheritance? What's the simplest interface that would support all use cases?